### PR TITLE
[WiP]Fix issue when onHMIStatus sent twice

### PR DESCRIPTION
**Pull request by okozlovlux**
- fixed issue when onHMIStatus sent twice

### CODE REVIEW

**Comment by okozlovlux** [comment]
Rebuild required

**Comment by okozlovlux** [comment]
@LevchenkoS @LuxoftAKutsan please reveiw

File: src/components/application_manager/include/application_manager/policies/policy_handler.h [hunk 1]
```diff
@@ -634,6 +634,9 @@ class PolicyHandler : public PolicyHandlerInterface,
 #ifdef SDL_REMOTE_CONTROL
   void UpdateHMILevel(application_manager::ApplicationSharedPtr app,
                       mobile_apis::HMILevel::eType level);
+
+  mobile_apis::DeviceRank::eType device_rank(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: @okozlovlux can it be const?

Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: @okozlovlux also please add description

Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: @okozlovlux PleaseRenameToGetDeviceRank. It is not getter.

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 2]
```diff
@@ -2241,5 +2241,25 @@ void PolicyHandler::OnUpdateHMILevel(const std::string& device_id,
...
+  const std::string device_id =
+      policy_manager_->GetCurrentDeviceId(application_id);
+  const Subject who = {device_id, application_id};
+
+  if (policy_manager_->access_remote()->IsAppReverse(who)) {
+    const std::string device_rank =
+        policy_manager_->access_remote()->IsPrimaryDevice(who.dev_id)
+            ? "DRIVER"
+            : "PASSENGER";
+    mobile_apis::DeviceRank::eType rank =
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/policies/policy_handler.cc]: @okozlovlux make it const or remove this variable.

File: src/components/policy/policy_regular/include/policy/policy_manager_impl.h [hunk 3]
```diff
@@ -381,6 +381,7 @@ class PolicyManagerImpl : public PolicyManager {
                                          const std::string& application_id);
   virtual bool GetModuleTypes(const std::string& policy_app_id,
                               std::vector<std::string>* modules) const;
+  virtual utils::SharedPtr<AccessRemote> access_remote();
```
Comment (LuxoftAKutsan) [inline src/components/policy/policy_regular/include/policy/policy_manager_impl.h]: @okozlovlux why virtual?

File: src/components/policy/policy_regular/src/policy_manager_impl.cc [hunk 4]
```diff
@@ -447,9 +447,6 @@ void PolicyManagerImpl::SendNotificationOnPermissionsUpdated(
 #ifdef SDL_REMOTE_CONTROL
   const Subject who = {device_id, application_id};
   if (access_remote_->IsAppReverse(who)) {
-    const std::string rank =
-        access_remote_->IsPrimaryDevice(who.dev_id) ? "DRIVER" : "PASSENGER";
-    UpdateDeviceRank(who, rank);
```
Comment (LuxoftAKutsan) [inline src/components/policy/policy_regular/src/policy_manager_impl.cc]: @okozlovlux isn't UpdateDEviceRank do anything except sending notification?

**Comment by okozlovlux** [comment]
@LuxoftAKutsan  @LevchenkoS  check lsat changes

**Comment by okozlovlux** [comment]
Not actual

**Pull request closed without merging**